src/bin/inx-chronicle/cli.rs

```diff
@@ -15,39 +15,90 @@ pub struct ClArgs {
     /// The location of the configuration file.
     #[arg(short, long, env = "CONFIG_PATH")]
     pub config: Option<String>,
-    /// The MongoDB connection string.
-    #[arg(long, env = "MONGODB_CONN_STR")]
-    pub mongodb_conn_str: Option<String>,
-    /// The url pointing to an InfluxDb instance.
-    #[arg(long, env = "INFLUXDB_URL")]
+    /// Rest API arguments.
+    #[cfg(feature = "api")]
+    #[command(flatten)]
+    pub api: ApiArgs,
+    /// InfluxDb arguments.
     #[cfg(feature = "influxdb")]
-    pub influxdb_url: Option<String>,
-    /// The address of the INX interface provided by the node.
-    #[arg(long, env = "INX_ADDR")]
+    #[command(flatten)]
+    pub influxdb: InfluxDbArgs,
+    /// INX arguments.
     #[cfg(feature = "inx")]
-    pub inx_addr: Option<String>,
-    /// Toggle INX write workflow.
-    #[arg(long, env = "INX")]
-    #[cfg(feature = "inx")]
-    pub enable_inx: Option<bool>,
+    #[command(flatten)]
+    pub inx: InxArgs,
+    /// Metrics arguments.
+    #[command(flatten)]
+    pub metrics: MetricsArgs,
+    /// MongoDb arguments.
+    #[command(flatten)]
+    pub mongodb: MongoDbArgs,
+    /// Subcommands.
+    #[command(subcommand)]
+    pub subcommand: Option<Subcommands>,
+}
+
+#[cfg(feature = "api")]
+#[derive(Args, Debug)]
+#[command(group = ArgGroup::new("api"))]
+pub struct ApiArgs {
+    /// Toggle REST API.
+    #[arg(long = "api-enabled", env = "REST_API_ENABLED", group = "api")]
+    pub api_enabled: Option<bool>,
+    /// JWT arguments.
+    #[command(flatten)]
+    pub jwt: JwtArgs,
+}
+
+#[derive(Args, Debug)]
+#[command(group = ArgGroup::new("jwt"))]
+pub struct JwtArgs {
     /// The location of the identity file for JWT auth.
-    #[arg(long, env = "JWT_IDENTITY_PATH")]
-    #[cfg(feature = "api")]
+    #[arg(long = "api-jwt-identity", env = "JWT_IDENTITY_PATH", group = "jwt")]
     pub identity_path: Option<String>,
     /// The password used for JWT authentication.
-    #[arg(long)]
-    #[cfg(feature = "api")]
+    #[arg(long = "api-jwt-password", group = "jwt")]
     pub password: Option<String>,
-    /// Toggle REST API.
-    #[arg(long, env = "API")]
-    #[cfg(feature = "api")]
-    pub enable_api: Option<bool>,
-    /// Toggle the metrics server.
-    #[arg(long, env = "METRICS")]
-    pub enable_metrics: Option<bool>,
-    /// Subcommands.
-    #[command(subcommand)]
-    pub subcommand: Option<Subcommands>,
+}
+
+#[cfg(feature = "inx")]
+#[derive(Args, Debug)]
+#[command(group = ArgGroup::new("inx"))]
+pub struct InxArgs {
+    /// Toggle INX write workflow.
+    #[arg(long = "inx-enabled", env = "INX_ENABLED", group = "inx")]
```

Comment:
Any reason why we need to specify the `long`s? They are automatically generated.

Reply:
remainder of the former "dot attempt". Sometimes it's necessary though as for the interface (aimed at the user) there might be something else more appropriate than just the converted field name. I'll change the places where it's the same.